### PR TITLE
First draft for file encryption simplify

### PR DESCRIPTION
**Pull request by ashutoshebi**
I have created the First draft. Creating pull request for reviewing files. No need to merge as discussed.

### CODE REVIEW

File: file-encryption-processor-simplify/src/main/java/uk/ac/ebi/ega/file/encryption/processor/configuration/KafkaConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        properties.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1);
+        properties.put(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, Integer.MAX_VALUE);
+        properties.put(ConsumerConfig.HEARTBEAT_INTERVAL_MS_CONFIG, 3000);
+        properties.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, 10000);
+        properties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest"); //TODO Change to latest
+        properties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
+        return properties;
+    }
+
+    @Bean
```
Comment (jorizci) [inline file-encryption-processor-simplify/src/main/java/uk/ac/ebi/ega/file/encryption/processor/configuration/KafkaConfiguration.java]: We can remove this Bean annotation

File: file-encryption-processor-simplify/src/main/java/uk/ac/ebi/ega/file/encryption/processor/configuration/KafkaConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        objectMapper.configure(SerializationFeature.WRITE_DATE_TIMESTAMPS_AS_NANOSECONDS, false);
+        return objectMapper;
+    }
+
+    @Bean
+    public ConsumerFactory<String, IngestionEventSimplify> consumerFactory() {
+        return new DefaultKafkaConsumerFactory<>(consumerConfigs());
+    }
+
+    @Bean
```
Comment (jorizci) [inline file-encryption-processor-simplify/src/main/java/uk/ac/ebi/ega/file/encryption/processor/configuration/KafkaConfiguration.java]: We can remove this bean annotation

File: file-encryption-processor-simplify/src/main/java/uk/ac/ebi/ega/file/encryption/processor/configuration/KafkaConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    public Map<String, Object> consumerConfigs() {
+        Map<String, Object> properties = new HashMap<>();
+        properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        // TODO either use a common package to handle this with versioning, user string + spring or Avro?
```
Comment (jorizci) [inline file-encryption-processor-simplify/src/main/java/uk/ac/ebi/ega/file/encryption/processor/configuration/KafkaConfiguration.java]: Remove the comment lines

File: file-encryption-processor-simplify/src/main/java/uk/ac/ebi/ega/file/encryption/processor/listener/FileEncryptionEventListener.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                                       final Path outputFolderPath) {
+        this.fileEncryptionProcessor = fileEncryptionProcessor;
+        this.kafkaTemplate = kafkaTemplate;
+        this.completeJobTopic = completeJobTopic;
+        this.outputFolderPath = outputFolderPath;
+    }
+
+    @SuppressWarnings("unchecked")
+    @KafkaListener(id = "${spring.kafka.client-id}", topics = "${spring.kafka.staging.ingestion.queue.name}",
+            groupId = "file-ingestion", clientIdPrefix = "executor", autoStartup = "false")
```
Comment (jorizci) [inline file-encryption-processor-simplify/src/main/java/uk/ac/ebi/ega/file/encryption/processor/listener/FileEncryptionEventListener.java]:
```suggestion
@KafkaListener(
            topics = "${file.encryption.queue.name}",
            groupId = "${file.encryption.group-id}",
            clientIdPrefix = "${file.encryption.instance-id}")
```
You can also remove the manual startup that it is configured on the original code and make it that it starts automatically.

File: file-encryption-processor-simplify/src/main/java/uk/ac/ebi/ega/file/encryption/processor/listener/FileEncryptionEventListener.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    public void listen(@Header(KafkaHeaders.RECEIVED_MESSAGE_KEY) String key, IngestionEventSimplify ingestionEventDataSimplify,
+                       Acknowledgment acknowledgment) {
+        logger.info("Process - key: {} data {}", key, ingestionEventDataSimplify);
+
+        final IIngestionEventData ingestionEventData = new IngestionEventData(ingestionEventDataSimplify, outputFolderPath);
+        final Result<ArchiveEventSimplify> result = fileEncryptionProcessor.encrypt(ingestionEventData);
+
+        /*Calling acknowledge() in both cases Success & Failure assuming FileManager will resend message
+        on kafka for re-encryption in case of failure.*/
+        acknowledgment.acknowledge();
```
Comment (jorizci) [inline file-encryption-processor-simplify/src/main/java/uk/ac/ebi/ega/file/encryption/processor/listener/FileEncryptionEventListener.java]: The other way around, be sure to have reported before acknowledging. If the report fails we can retry (or wait here until the report is properly sent) Worst case scenario the manager has been reported that the process finished and the executor does the work again and sends a duplicated message to the file manager. That is something the manager needs to control.

File: file-encryption-processor/src/main/java/uk/ac/ebi/ega/file/encryption/processor/listener/IngestionEventListener.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+import uk.ac.ebi.ega.file.encryption.processor.services.IEncryptService;
+import uk.ac.ebi.ega.ingestion.commons.messages.IngestionEvent;
+import uk.ac.ebi.ega.jobs.core.JobExecution;
+import uk.ac.ebi.ega.jobs.core.Result;
+import uk.ac.ebi.ega.jobs.core.exceptions.JobNotRegistered;
+
+import java.util.Optional;
+
+@Component
+public class IngestionEventListener {
```
Comment (jorizci) [inline file-encryption-processor/src/main/java/uk/ac/ebi/ega/file/encryption/processor/listener/IngestionEventListener.java]: Delete?

File: file-encryption-processor-simplify/src/main/java/uk/ac/ebi/ega/file/encryption/processor/listener/FileEncryptionEventListener.java (hunk 4 above)
Comment (ashutoshebi) [reply file-encryption-processor-simplify/src/main/java/uk/ac/ebi/ega/file/encryption/processor/listener/FileEncryptionEventListener.java]: Done

File: file-encryption-processor-simplify/src/main/java/uk/ac/ebi/ega/file/encryption/processor/listener/FileEncryptionEventListener.java (hunk 5 above)
Comment (ashutoshebi) [reply file-encryption-processor-simplify/src/main/java/uk/ac/ebi/ega/file/encryption/processor/listener/FileEncryptionEventListener.java]: Done

File: file-encryption-processor-simplify/src/test/java/uk/ac/ebi/ega/file/encryption/processor/pipeline/DefaultIngestionPipelineTest.java [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    private String outputFolderPath;
+
+    @Value("${file.to.encrypt.path}")
+    private String fileToEncryptPath;
+
+    @Value("${file.plain.md5.path}")
+    private String originalPlainFileMD5Path;
+
+    @After
+    public void cleanTestEnvironment() {
```
Comment (jorizci) [inline file-encryption-processor-simplify/src/test/java/uk/ac/ebi/ega/file/encryption/processor/pipeline/DefaultIngestionPipelineTest.java]: Not needed

**Pull request merged**